person.rb

```diff
@@ -0,0 +1,39 @@
+require_relative 'capitalize_decorator'
+require_relative 'trimmer_decorator'
+require_relative 'nameable'
+
+# for person
+class Person < Nameable
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def initialize(id, age, name: 'Unknown', parent_permission: true)
+    super()
+    @id = id
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  def correct_name
+    @name
+  end
+end
+
+person1 = Person.new(1, 17, name: 'Abutalha', parent_permission: false)
+person2 = Person.new(2, 16, name: 'Rex')
+person3 = Person.new(3, 35, name: 'Unknown', parent_permission: true)
+
+puts person1.can_use_services? # false
+puts person2.can_use_services? # true
+puts person3.can_use_services? # true
```

Comment:
-  Please kindly double-check :heavy_check_mark:  if you can decorate a person's object as requested in the requirements. Kindly create a person's object with a name and verify if you can actually decorate the name(correct, capitalize, and trim) using respective decorators classes. Right now the person object doesn't even have access to the correct_name method because it is private 
![Screenshot from 2023-11-10 20-48-20](https://github.com/AbuTalha3/school-library-oop/assets/63087491/ec55f82f-f562-48ae-a4f4-d9fdd88bbd73)